src/xkbcomp/rules.c

```diff
@@ -343,6 +350,39 @@ matcher_group_add_element(struct matcher *m, struct sval element)
                   element);
 }
 
+static int
+read_rules_file(struct xkb_context *ctx,
+                struct matcher *matcher,
+                const char *path,
+                struct xkb_component_names *out);
+static void
+matcher_include(struct matcher *m, struct sval inc,
+                struct xkb_component_names *out)
+{
+    char *path = strndup(inc.start, inc.len);
+    bool ret;
+
+    /* %h at the beginning expands to $HOME */
+    if (strlen(path) > 2 && strncmp(path, "%h", 2) == 0) {
```

Comment:
If we interpret `%`-expansions, I'd prefer that `%` be matched separately, and if the following character is not recognized, a warning be issued. We should also allow a `%%` to expand to `%`, I suppose.

The Compose parser [does this](https://github.com/xkbcommon/libxkbcommon/blob/934d5741727886fe90e9c2a4d90e63bbd2cb2226/src/compose/parser.c#L258-L296).